web/com.etendoerp.copilot.chat/src/App.tsx

```diff
@@ -150,56 +153,64 @@ function App() {
     setFileId(null);
 
     if (!isBotLoading) {
-      const question = inputValue.trim();
+      const originalQuestion = inputValue.trim();
       setInputValue('');
-      if (!question) return;
+      if (!originalQuestion) return;
+
+      let finalQuestion = originalQuestion;
+
+      if (isFirstMessage) {
+        setIsFirstMessage(false);
+        if (contextValue) {
+          finalQuestion = `Context: ${contextValue}\nQuestion: ${originalQuestion}`;
+        }
+      }
 
       // Add user message
       const userMessage: IMessage = {
-        text: question,
+        text: originalQuestion,
         sender: ROLE_USER,
         timestamp: formatTimeNewDate(new Date()),
       };
+
       if (file) {
         userMessage.file = file.name;
       }
+
       await handleNewMessage(ROLE_USER, userMessage);
       setStatusIcon(botIcon);
```

Comment:
Having a set state after an await could lead to a setState in a unmounted component. Probably you already know how to use it but using an AbortController would be safer:
https://dev.to/collegewap/react-prevent-state-updates-on-unmounted-components-2j4d